core/src/main/java/com/linecorp/armeria/server/file/AbstractHttpFileBuilder.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    /**
+     * Sets whether to set the {@code "last-modified"} header automatically. By default,
+     * the {@code "last-modified"} is set automatically.
+     */
+    public final B lastModified(boolean lastModifiedEnabled) {
+        this.lastModifiedEnabled = lastModifiedEnabled;
+        return self();
+    }
+
+    /**
+     * Sets whether to set the {@code "content-type"} header automatically based on the extension of the file.
+     */
+    protected final boolean isContentTypeAutoDetectionEnabled() {
+        return contentTypeAutoDetectionEnabled;
+    }
+
+    /**
+     * Sets whether to set the {@code "content-type"} header automatically based on the extension of the file.
+     * By default, the {@code "content-type"} header is set automatically.
+     */
+    public final B autoDetectedContentType(boolean contentTypeAutoDetectionEnabled) {
+        this.contentTypeAutoDetectionEnabled = contentTypeAutoDetectionEnabled;
+        return self();
+    }
+
+    /**
+     * Returns the function which generates the entity tag that's used for setting the {@code "etag"} header
+     * automatically.
+     *
+     * @return the etag function, or {@code null} if the {@code "etag"} header is not set automatically.
+     */
+    @Nullable
+    protected final BiFunction<String, HttpFileAttributes, String> entityTagFunction() {
+        return entityTagFunction;
+    }
+
+    /**
+     * Sets whether to set the {@code "etag"} header automatically based on the path and attributes of the
+     * file. By default, the {@code "etag"} header is set automatically. Use {@link #entityTag(BiFunction)} to
+     * customize how an entity tag is generated.
+     */
+    public final B entityTag(boolean enabled) {
+        entityTagFunction = enabled ? DefaultEntityTagFunction.get() : null;
+        return self();
+    }
+
+    /**
+     * Sets the function which generates the entity tag that's used for setting the {@code "etag"} header
+     * automatically.
+     *
+     * @param entityTagFunction the entity tag function that generates the entity tag, or {@code null}
+     *                          to disable setting the {@code "etag"} header.
+     */
+    public final B entityTag(BiFunction<String, HttpFileAttributes, String> entityTagFunction) {
+        this.entityTagFunction = requireNonNull(entityTagFunction, "entityTagFunction");
+        return self();
+    }
+
+    /**
+     * Returns the immutable additional {@link HttpHeaders} which will be set when building an
+     * {@link HttpResponse}.
+     */
+    protected final HttpHeaders headers() {
+        return headers != null ? headers.asImmutable() : HttpHeaders.EMPTY_HEADERS;
+    }
+
+    private HttpHeaders getOrCreateHeaders() {
+        if (headers == null) {
+            headers = HttpHeaders.of();
+        }
+        return headers;
+    }
+
+    /**
+     * Adds the specified HTTP header.
+     */
+    public final B addHeader(CharSequence name, Object value) {
```

Comment:
I was looking at using the new http file methods but realized for a resource directory (serving many files, not just one) we still use `HttpFileServiceBuilder` but don't have convenient methods like `addHeader`. Does it make sense to add these helpers to `HttpFileServiceBuilder` too?

Reply:
We have them already in `master`. Just need to release this week. :wink:

Reply:
Ah sweet!